### PR TITLE
instructorCourseDelete may not be scalable #2973

### LINKED ISSUE #2973: instructorCourseDelete may not be scalable
Check if this operation is salable for large courses with many responses

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #2973

### CODE REVIEW

**Comment by kanghj** [comment]
Deleting a course created by the admin for a new instructor,
This is the graph on app stats before these changes (on master):

before

This is the graph on app stats after the changes (on this branch):

after

**Comment by wkurniawan07** [comment]
I hope this gets merged quick. The failing AHPUiT is really getting on my nerves.

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 1]
```diff
@@ -1527,12 +1527,12 @@ public void scheduleFeedbackSessionPublishedEmails() {
...
-
-        for (FeedbackSessionAttributes session : sessionsToDelete) {
-            deleteFeedbackSessionCascade(session.feedbackSessionName, session.courseId);
-        }
+        frcLogic.deleteFeedbackResponseCommentsForCourse(courseId);
+        frLogic.deleteFeedbackResponsesForCourse(courseId);
+        fqLogic.deleteFeedbackQuestionsForCourse(courseId);
+        
+        List<FeedbackSessionAttributes> fsList = getFeedbackSessionsForCourse(courseId);
+        fsDb.deleteEntities(fsList);
```
Comment (LowWeiLin) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: Maybe renaming/wrapping deleteEntities would make what it does clearer when called from outside fsDb.

**Comment by kanghj** [comment]
Changes made @LowWeiLin

**Comment by kanghj** [comment]
I think I will need to make more changes to this. One thing I noticed is that the methods I switched to using were used by the BackDoor to delete the courses before persisting test data to the datastore. One difference is that these methods doesn't do <code>log.info(entityToDelete.getBackupIdentifier())</code>. @damithc, is it expected behaviour that this backdoor operations doesn't log when deleting data?

**Comment by junhaoyap** [comment]
Was wondering why admin test is still failing, apparently cause this hasn't been merged :'(

**Comment by damithc** [comment]
Logging of entity changes was done as part of a data backup mechanism that we never used in the end. So it might be arbitrarily done at this point, but it's not supposed to be like that. The idea was that by examining logs we should be able to identify which entities changed during a specific period  so that we can create an incremental back up of those entities. For now, we can ignore that requirement.

**Pull request merged**